### PR TITLE
Fixes#511  Transform the quick fixes which use ImplementInterfaceProposal to implement the code action participant extension point. 

### LINKED ISSUE #511: Transform the quick fixes which use ImplementInterfaceProposal to implement the code action participant extension point.
See the initial work completed here: https://github.com/OpenLiberty/liberty-tools-intellij/issues/309.

These quick fixes rely upon ImplementInterfaceProposal:
- [x] FilterImplementationQuickFix
- [x] ListenerImplementationQuickFix

In addition to refactoring the code, the code action participants will need to be added to the plugin.xml and the relevant Jakarta EE code action tests should be enabled.

### PR DESCRIPTION
**Pull request by aparnamichael**
Fixes #511 
Refactored the following quick fixes :

 1. FilterImplementationQuickFix
2.  ListenerImplementationQuickFix

These PR contains changes from Dessina's fork as well.

### CODE REVIEW

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/servlet/FilterImplementationQuickFix.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2020, 2023 Red Hat Inc. and others.
+ * Copyright (c) 2020, 2023, 2024 Red Hat Inc. and others.
```
Comment (vaisakhkannan) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/servlet/FilterImplementationQuickFix.java:2]: Year should be like this "2020, 2024"
"Start year, latest update year"

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java
Comment (vaisakhkannan) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java:1]: Copyright (c) 2019, 2024 Red Hat, Inc. and others

The header year should be as shown above.

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java [hunk 2]
```diff
@@ -128,12 +126,6 @@ public List<CodeAction> codeAction(JakartaJavaCodeActionParams params, IPsiUtils
                     if (diagnostic.getCode().getLeft().equals(ServletConstants.DIAGNOSTIC_CODE)) {
                         codeActions.addAll(HttpServletQuickFix.getCodeActions(context, diagnostic));
                     }
-                    if (diagnostic.getCode().getLeft().equals(ServletConstants.DIAGNOSTIC_CODE_FILTER)) {
```
Comment (vaisakhkannan) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java:131]: Why we removing this?

**Review by vaisakhkannan — APPROVED** [review]
Looks Good for me

**Review by mrglavas — APPROVED** [review]
Looks good. Thanks for opening https://github.com/eclipse/lsp4jakarta/issues/521 to cover adding the new tests to LSP4Jakarta.

**Pull request merged**